domain/machine/service/watcher.go

```diff
@@ -75,6 +75,20 @@ func (s *WatchableService) WatchMachineCloudInstances(ctx context.Context) (watc
 	)
 }
 
+// WatchMachineReboot returns a StringsWatcher that is subscribed to
+// the changes in the machine_requires_reboot table in the model.
+// It includes in its change stream the UUID of machine which have been requested for reboot and
+// their potential children.
+func (s *WatchableService) WatchMachineReboot(ctx context.Context) (watcher.StringsWatcher, error) {
+	table, stmt := s.st.InitialWatchMachineRebootStatement()
+	return s.watcherFactory.NewNamespaceMapperWatcher(
+		table,
+		changestream.Create,
+		eventsource.InitialNamespaceChanges(stmt),
+		expandWithChildMapper,
+	)
+}
```

Comment:
The watch machine reboot API in the state package and at the API server package use a NotifyWatcher. It allows for coalescing at the database level.

Reply:
Yeah, then you don't need to do the initial statement thing.

Reply:
I am not sure to understand this comment, neither its intent.

Through the code, I can find several interfaces named NotifyWatcher, but neither of them allows to return any values (no UUID). I have seen that in state package, a NotifyWatcher was used, but for me it was because a watcher was attached to each machine, so the information of which machine ask for reboot was already shipped by the identity of the watcher.

As far I understand the actual implementation (and thus, probably actual design) of machine domain, we use watchers for whole machine domain events, and we will have only have one watcher for all machines, so the watcher requires to yield machines identities (UUID).

I didn't challenge this idea because it feels more scalable to have one watcher watching all reboot event than having n watcher watching n machine for reboot events (I see more opportunities and advantages this way than the other way).

So, what did you mean through this comment ?

Reply:
got it: we shouldn't a have a watcher which watch them all (not the One Watcher), but a watcher by machine.)

refactored.